projects/postgresql/fuzzer/fuzzer_initialize.c

```diff
@@ -0,0 +1,112 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+///////////////////////////////////////////////////////////////////////////////
+
+#include "postgres.h"
+
+#include "access/xlog.h"
+#include "access/xact.h"
+#include "common/username.h"
+#include "executor/spi.h"
+#include "jit/jit.h"
+#include "libpq/libpq.h"
+#include "libpq/pqsignal.h"
+#include "miscadmin.h"
+#include "optimizer/optimizer.h"
+#include "parser/analyze.h"
+#include "parser/parser.h"
+#include "storage/proc.h"
+#include "tcop/tcopprot.h"
+#include "utils/datetime.h"
+#include "utils/memutils.h"
+#include "utils/portal.h"
+#include "utils/snapmgr.h"
+#include "utils/timeout.h"
+
+const char *progname;
+static MemoryContext row_description_context = NULL;
+static StringInfoData row_description_buf;
+static const char *username = NULL;
+
+static void fuzzer_exit(){
+  if(username)
+    pfree((void *) username);
+}
+
+int FuzzerInitialize(char *dbname){
+  char *argv[5];
+  char arg_path[50];
+  int arg_path_size;
+  char path_to_db[50];
+  int path_to_db_size;
+  char untar[100];
+  int untar_size;
+  arg_path_size = snprintf(arg_path, 50, "/tmp/%s/data", dbname);
+  path_to_db_size = snprintf(path_to_db, 50, "-D\"/tmp/%s/data\"", dbname);
+  untar_size = snprintf(untar, 100, "rm -rf /tmp/%s && mkdir /tmp/%s && tar -xvf data.tar.gz -C /tmp/%s", dbname, dbname, dbname);
+  
+  argv[0] = "tmp_install/usr/local/pgsql/bin/postgres";
+  argv[1] = path_to_db;
+  argv[2] = "-F";
+  argv[3] = "-k\"/tmp/pg_dbfuzz\"";
+  argv[4] = NULL;
+
+  system(untar);
+  
+  progname = get_progname(argv[0]);
+  MemoryContextInit();
+
+  username = strdup(get_user_name_or_exit(progname));
+	 
+  InitStandaloneProcess(argv[0]);
+  SetProcessingMode(InitProcessing);
+  InitializeGUCOptions();
+  process_postgres_switches(4, argv, PGC_POSTMASTER, &dbname);
+  dbname = "dbfuzz";
```

Comment:
if you are not interested in the process_postgres_switches giving you back the db name, then give it NULL instead of &dbname